crates/sncast/src/starknet_commands/verification/voyager.rs

```diff
@@ -0,0 +1,52 @@
+use super::base::{BaseVerificationInterface, VerificationInterface, VerificationPayload};
+use anyhow::Result;
+use async_trait::async_trait;
+use camino::Utf8PathBuf;
+use sncast::response::structs::VerifyResponse;
+use sncast::Network;
+use starknet::core::types::FieldElement;
+use std::env;
+
+pub struct VoyagerVerificationInterface {
+    base: BaseVerificationInterface,
+}
+
+#[async_trait]
+impl VerificationInterface for VoyagerVerificationInterface {
+    fn new(network: Network, workspace_dir: Utf8PathBuf) -> Self {
+        VoyagerVerificationInterface {
+            base: BaseVerificationInterface {
+                network,
+                workspace_dir,
+            },
+        }
+    }
+
+    async fn verify(
+        &self,
+        contract_address: Option<FieldElement>,
+        class_hash: Option<FieldElement>,
+        class_name: String,
+    ) -> Result<VerifyResponse> {
+        let file_data = self.base.read_workspace_files()?;
+        let source_code = serde_json::Value::Object(file_data);
+        let payload = VerificationPayload {
+            class_name,
+            contract_address,
+            class_hash,
+            source_code,
+        };
+        let url = self.gen_explorer_url()?;
+        self.base.send_verification_request(url, payload).await
+    }
+
+    fn gen_explorer_url(&self) -> Result<String> {
+        let api_base_url = env::var("VOYAGER_API_URL")
```

Comment:
We would like to unify the wole configuration to a TOML file. Could you please redesign the url acquisition logic to use an enum provied via `CastConfig` rather than an environmental variable?